applications/data_generator.py

```diff
@@ -334,30 +335,108 @@ def worker(h_idx, config=None, part_per_holo=None, empt_per_holo=None):
+#             for data in tqdm.tqdm(p.imap(work,
+#                                          test_hologram_numbers), total=total_testing_examples):
+#                 for image, label, mask in zip(data["stacked_image"], data["label"], data["mask"]):
+#                     joblib.dump((image, label, mask), fid)
+                    
+        # Training split
+        total_training_examples = 10
+        
+        
+        X = np.zeros((total_training_examples, 2, tile_size, tile_size), dtype = np.float32)
+        Y1 = np.zeros((total_training_examples, tile_size, tile_size), dtype = np.int)
+        Y2 = np.zeros((total_training_examples, 1), dtype = np.int)
+        c = 0
+        for data in tqdm.tqdm(p.imap(work,
+                                     training_hologram_numbers), total=total_training_examples):
+            for image, label, mask in zip(data["stacked_image"], data["label"], data["mask"]):
+                X[c:c+1] = image
+                Y1[c:c+1] = mask.toarray()
+                Y2[c:c+1] = np.array([int(label)])
+                c += 1
+
+                if c >= total_training_examples:
+                    break
+            if c >= total_training_examples:
+                break
+        
+        df = xr.Dataset(data_vars=dict(var_x=(['n', 'd', 'x', 'y'], X[:c]),
+                                       var_y=(['n', 'x', 'y'], Y1[:c]),
+                                       var_z=(['n', 'z'], Y2[:c])))
+        
+        df.to_netcdf("/glade/work/schreck/repos/HOLO/033022/holodec-ml/results/TEST/test.nc")
+        #df.to_netcdf(f"{output_path}/training_{name_tag}.nc")
+        raise
+        
+        
+        # Validation split
+        X = np.zeros((total_validation_examples, 2, tile_size, tile_size), dtype = np.float32)
+        Y1 = np.zeros((total_validation_examples, tile_size, tile_size), dtype = np.int)
+        Y2 = np.zeros((total_validation_examples, 1), dtype = np.int)
+        c = 0
+        for data in tqdm.tqdm(p.imap(work,
+                                     validation_hologram_numbers), total=total_validation_examples):
+            for image, label, mask in zip(data["stacked_image"], data["label"], data["mask"]):
+                X[c:c+1] = image
+                Y1[c:c+1] = mask.toarray()
+                Y2[c:c+1] = np.array([int(label)])
+                c += 1
+
+                if c >= total_validation_examples:
+                    break
+            if c >= total_validation_examples:
+                break
+        
+        df = xr.Dataset(data_vars=dict(var_x=(['n', 'd', 'x', 'y'], X[:c]),
+                                       var_y=(['n', 'x', 'y'], Y1[:c]),
+                                       var_z=(['n', 'z'], Y2[:c])))
+        
+        df.to_netcdf(f"{output_path}/validation_{name_tag}.nc")
+        
+        # Test split
+        X = np.zeros((total_testing_examples, 2, tile_size, tile_size), dtype = np.float32)
```

Comment:
PEP 8 note: according to the Python style guide, there are no spaces around `=` in keyword arguments, such as `dtype=np.float32`. There are spaces are `=` for variable assignment.

Reply:
Ah; this happens in a number of scripts, will track them down before the merge.